src/js/components/Button/__tests__/Button-kind-test.js

```diff
@@ -300,6 +300,42 @@ describe('Button kind', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
+  // Custom theme with intelligentPad set to true
+  const themeWithIntelligentPadTrue = {
+    button: {
+      size: {
+        medium: {
+          pad: {
+            horizontal: '10px',
+            vertical: '10px',
+          },
+        },
+      },
+      intelligentPad: true,
+      default: {},
+      secondary: {
+        border: {
+          color: 'brand',
+          width: '2px',
+        },
+        color: 'text-strong',
+        font: {
+          weight: 600,
+        },
+      },
+    },
+  };
+
+  test('padding is applied correctly when intelligentPad is true', () => {
```

Comment:
Can we also have a test where intelligentPad is false?

Reply:
added